src/webapp/components/user-notification/useNotificationDetailsDialog.ts

```diff
@@ -0,0 +1,89 @@
+import { useAppContext } from "../../contexts/app-context";
+import { useCallback, useState } from "react";
+import { NotificationAttrs, NotificationWildcard } from "../../../domain/entities/Notification";
+import { ShareUpdate } from "@eyeseetea/d2-ui-components";
+import { NotificationDetailsDialogProps } from "./NotificationDetailsDialog";
+import { generateUid } from "../../../data/utils/uid";
+import { NamedRef } from "../../../domain/entities/Ref";
+import { SharingRule } from "@eyeseetea/d2-ui-components/sharing/types";
+
+export const useNotificationDetailsDialog = (props: NotificationDetailsDialogProps) => {
+    const { onSave, initialNotification } = props;
+    const { compositionRoot } = useAppContext();
+    const [notification, setNotification] = useState(initialNotification || newNotification());
+
+    const handleSave = useCallback(async () => {
+        await onSave(notification);
+    }, [notification, onSave]);
+
+    const onContentChange = useCallback((content: string) => {
+        setNotification(notification => ({ ...notification, content }));
+    }, []);
+
+    const onSharingChanged = useCallback(async (updatedAttributes: ShareUpdate) => {
+        setNotification(notification => {
+            const { users, userGroups, wildcard } = notification.recipients;
+            const { userAccesses, userGroupAccesses } = updatedAttributes;
+            return {
+                ...notification,
+                recipients: {
+                    users: userAccesses ? mapSharingRulesToRecipient(userAccesses) : users,
+                    userGroups: userGroupAccesses ? mapSharingRulesToRecipient(userGroupAccesses) : userGroups,
+                    wildcard: wildcard,
+                },
+            };
+        });
+    }, []);
+
+    const onWildcardChange = useCallback(value => {
+        setNotification(notification => ({
+            ...notification,
+            recipients: {
+                ...notification.recipients,
+                wildcard: value,
+            },
+        }));
+    }, []);
+
+    const searchSharing = useCallback(
+        (query: string) => compositionRoot.instance.searchUsers(query),
+        [compositionRoot]
+    );
+
+    return {
+        handleSave,
+        notification,
+        onContentChange,
+        onSharingChanged,
+        onWildcardChange,
+        searchSharing,
+        sharingMeta: generateSharingMeta(notification),
+    };
+};
+
+function newNotification(): NotificationAttrs {
```

Comment:
NotificationAttrs is the object that Notification.create takes, and since that's a public method, it's ok to export it. But we should not typically use it. Our real thing here is the "Notification", we should work with this object.